beancount/parser/tokens.h

```diff
@@ -12,19 +12,22 @@
  * are handled and reported as lexing errors.
  */
 #define token(name, ...)                                        \
-    yylval->pyobj = build_##name(__VA_ARGS__);                  \
+    yylval->pyobj = invoke_builder(name, (__VA_ARGS__));        \
     if (yylval->pyobj == NULL) {                                \
         build_lexer_error_from_exception(yylloc, builder);      \
         return YYerror;                                         \
     }
 
+/* MSVC requires this indirection to expand __VA_ARGS__ correctly. */
+#define invoke_builder(name, arg_list) build_##name arg_list
+
 #define build_STR(_ptr, _len) PyUnicode_FromStringAndSize(_ptr, _len)
 #define build_KEY build_STR
 #define build_TAG build_STR
 #define build_LINK build_STR
 #define build_CURRENCY build_STR
 #define build_BOOL(_value) PyBool_FromLong(_value)
-#define build_NONE() ({ Py_INCREF(Py_None); Py_None; })
+#define build_NONE() ( Py_INCREF(Py_None), Py_None )
```

Comment:
I don't think this is correct.

Reply:
Why?

Reply:
@dnicolodi It's corret, it builds fine after this patch is apply on master.

Reply:
It works, indeed. I don't know what I was thinking when I read the code the first time.